files/zh-cn/web/javascript/reference/global_objects/string/index.md

```diff
@@ -152,59 +152,22 @@ console.log(eval(s2.valueOf())); // returns the number 4
-| `\b`                                                                                                                                       | 退格键（U+0008 BACKSPACE）                                                                          |
-| `\f`                                                                                                                                       | 换页符（U+000C FORM FEED）                                                                          |
-| `\uXXXX` …其中 `XXXX` 恰好是 `0000`–`FFFF` 范围内的 4 个十六进制；例如，`\u000A` 与 `\n`（换行）相同；`\u0021` 是 `!`                      | `U+0000` 和 `U+FFFF` 之间的 Unicode 码位（Unicode 基于多平台语言）                                  |
-| `\u{X}`…`\u{XXXXXX}` …其中 `X`…`XXXXXX` 是 `0`–`10FFFF` 范围内的 1-6 个十六进制数字；例如，`\u{A}` 与 `\n`（换行符）相同； `\u{21}` 是 `!` | `U+0000` 和 `U+10FFFF` 之间的 Unicode 码位（整个 Unicode）                                          |
-| `\xXX` …其中 `XX` 恰好是 `00`–`FF` 范围内的 2 个十六进制数字；例如，`\x0A` 与 `\n`（换行符）相同；`\x21` 是 `!`                            | `U+0000` 和 `U+00FF` 之间的 Unicode 码位（Basic Latin 和 Latin-1 Supplement 块；相当于 ISO-8859-1） |
-
-### 长字符串文字
-
-有时，你的代码将包含非常长的字符串。你可能希望在不影响实际字符串内容的情况下，在源代码中专门将字符串分成很多行，而不是无休止的继续下去，或者随编辑器的心血来潮换行。
-
-你可以使用 [`+`](/zh-CN/docs/Web/JavaScript/Reference/Operators/Addition) 运算符去增加多行字符串，像这样：
+### UTF-16 字符、Unicode 码位和字素簇
 
-```js
-const longString =
-  "This is a very long string which needs " +
-  "to wrap across multiple lines because " +
-  "otherwise my code is unreadable.";
-```
-
-或者你可以在每行的末尾使用反斜杠字符（`\`）来指示该字符串将在下一行继续。确保没有空格或者任何其他字符在反斜杠或缩进之后，否则它将不工作。
-
-```js
-const longString =
-  "This is a very long string which needs \
-to wrap across multiple lines because \
-otherwise my code is unreadable.";
-```
+字符串基本上表示为 [UTF-16 码元](https://zh.wikipedia.org/wiki/UTF-16)的序列。在 UTF-16 编码中，每个码元都是 16 位长。这意味着最多有 2<sup>16</sup> 个或 65536 个可能的字符可表示为单个 UTF-16 码元。该字符集称为[基本多语言平面（BMP）](https://zh.wikipedia.org/wiki/Unicode字符平面映射)，包含最常见的字符，如拉丁字母、希腊字母、西里尔字母以及许多东亚字符。每个码元都可以用以 `\u` 开头的 4 个十六进制数字写在一个字符串中。
 
-上述两种方法都会产生相同的字符串。
+然而，整个 Unicode 字符集比 65536 大得多。额外的字符以*代理对*（surrogate pair）的形式存储在 UTF-16 中，代理对是一对 16 位码元，表示一个单个字符。为了避免起义，配对的两个部分必须介于 `0xD800` 和 `0xDFFF` 之间，并且这些码元不用于编码单码元字符。（更准确地说，前导代理的值在 `0xD800` 和 `0xDBFF` 之间（含），而后尾代理的值在 `0xDC00` 和 `0xDFFF` 之间（含）。）每个 Unicode 字符由一个或者两个 UTF-16 码元组成，也称为 _Unicode 码位_（code point）。每个 Unicode 码位都可以使用 `\u{xxxxxx}` 写成一个字符串，其中 `xxxxxx` 表示 1–6 个十六进制数字。
 
-### UTF-16 字符、Unicode 码位和字素簇（grapheme clusters）
+“单独代理项”是指满足以下描述之一的 16 位码元：
 
-字符串基本上表示为 [UTF-16 码元](https://zh.wikipedia.org/wiki/UTF-16)的序列。在 UTF-16 编码中，每个码元都是 16 位长。这意味着最多有 2<sup>16</sup> 个或 65536 个可能的字符可表示为单个 UTF-16 码元。该字符集称为[基本多语言平面（BMP）](<https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane>)，包含最常见的字符，如拉丁字母、希腊字母、西里尔字母以及许多东亚字符。每个码元都可以写成一个字符串，其中 `\u` 之后跟四个十六进制数字。
+- 它在范围 `0xD800` 到 `0xDBFF` 内（含）（即为前导代理），但它是字符串中的最后一个码元，或者下一个码元不是后尾代理。
+- 它在范围 `0xDC00` 到 `0xDFFF` 内（含）（即为后尾代理），但它是字符串中的第一个码元，或者前一个码元不是前导代理。
```

Comment:
```suggestion
- 它在范围 `0xD800` 到 `0xDBFF` 内（含）（即为高代理项），但它是字符串中的最后一个码元，或者下一个码元不是低代理项。
- 它在范围 `0xDC00` 到 `0xDFFF` 内（含）（即为低代理项），但它是字符串中的第一个码元，或者前一个码元不是高代理项。
```

Reply:
according to wiki, refer: https://zh.wikipedia.org/zh-hans/UTF-16

> 高位的10比特的值（值的范围为 0...0x3FF）被加上 0xD800 得到第一个码元或称作高位代理（high surrogate），值的范围是 0xD800...0xDBFF。由于高位代理比低位代理的值要小，所以为了避免混淆使用，Unicode标准现在称高位代理为前导代理（lead surrogates）。
> 低位的10比特的值（值的范围也是 0...0x3FF）被加上 0xDC00 得到第二个码元或称作低位代理（low surrogate），现在值的范围是 0xDC00...0xDFFF。由于低位代理比高位代理的值要大，所以为了避免混淆使用，Unicode标准现在称低位代理为后尾代理（trail surrogates）。

maybe we should update the upstream